src/pages/ActivationPage.tsx

```diff
@@ -16,37 +18,45 @@ const CodeBox = styled.div`
   margin: auto;
 `
 
-// TODO: Codes will eventually be derived from the election file.
-const validBallotCodes = [
-  'VX.precinct-23.12D',
-  'VX.precinct-21.5R',
-  'MyVoiceIsMyPassword',
-  'Foo',
-  'Password',
-  'VotingWorks',
-]
-
 let resetBallotCode: number
 
 const StartPage = (props: RouteComponentProps) => {
-  const { setBallotKey } = useContext(BallotContext)
+  const { election: contextElection, activateBallot } = useContext(
+    BallotContext
+  )
+  const election = contextElection as Election
   const [ballotCode, setBallotCode] = useState('')
-  const setBallot = (code: string) => {
+  const setBallot = (activationData: ActivationData) => {
     clearTimeout(resetBallotCode)
-    setBallotKey(code)
+    activateBallot(activationData)
     props.history.push('/start')
   }
 
-  /* istanbul ignore next */
+  /* istanbul ignore next - shortcut will not exist in official release */
   const takeShortcut = () => {
-    setBallot(validBallotCodes[0])
+    const ballotStyle = election.ballotStyles[0]
+    setBallot({
+      ballotStyle,
+      precinct: election.precincts[0],
+    })
   }
   const onSubmit = (event: React.FormEvent) => {
     event.preventDefault()
+    const [brand, precinctId, ballotStyleId] = ballotCode.split('.')
```

Comment:
my preference, in these situations where a critical security feature has a few lines of logic, is to make that a function with a clear name, even if it's only used once. Otherwise, it's hard to tell what's security critical and what is not.

Reply:
Ok, done did dis ding.